src/views/AnnualReport.vue

```diff
@@ -859,9 +869,19 @@ export default {
           }
           filing = res.data.filing
           this.haveChanges = false
-        }).catch(error => {
+        }).catch(async error => {
           if (error && error.response && error.response.status === PAYMENT_REQUIRED) {
-            this.paymentErrorDialog = true
+            if (error.response.data && error.response.data.errors) {
+              const msgCode = error.response.data.errors.find(x => x.payment_error_type.startsWith('BCOL'))
```

Comment:
Just a question, will we always looking for BCOL? is it possible we have other payment error types and we want to just display it if there is a payment error?

Reply:
Yup, these are BCOL specific errors, and they all start with BCOL. I don't know if we have a future plan to store/retrieve all payment error HTML on the Pay API side

Reply:
Since Ody is away this week, I'll say this looks satisfactorily handled and am closing this conversation.